lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -0,0 +1,137 @@
+defmodule ArchEthicWeb.FaucetController do
+  @moduledoc false
+
+  alias ArchEthic.TransactionChain.{
+    Transaction,
+    TransactionData,
+    TransactionData.Ledger,
+    TransactionData.UCOLedger
+  }
+
+  alias ArchEthic.Crypto
+
+  use ArchEthicWeb, :controller
+
+  def index(conn, __params) do
+    conn
+    |> put_resp_header("cache-control", "no-cache, no-store, must-revalidate")
+    |> put_resp_header("pragma", "no-cache")
+    |> put_resp_header("expires", "0")
+    |> render("index.html", address: "")
+  end
+
+  def create_transfer(conn, %{"address" => address}) do
+    case transfer(address, :secp256r1) do
+      :ok ->
+        conn
+        |> put_flash(:info, "Transferred successfully")
+        |> redirect(to: "/faucet")
+
+      {:error, _} ->
+        conn
+        |> put_flash(:error, "Unable to transfer")
+        |> render("index.html", address: address)
+    end
+  end
+
+  def transfer(
```

Comment:
I think you should put it in private as it's only used by the controller function.